### ISSUE TITLE
Include an __eq__ method for Embeds (v.2)

### ISSUE THREAD

**Issue by dob9601** [Rapptz/discord.py#5962]
## The Problem

2 embeds that are identical won't necessarily be flagged as equal when passed through an equivalence check. However, the dictionaries returned by their to_dict methods do.

## The Ideal Solution

Add the following method or something to the same effect to the Embed class:
```py
def __eq__(self, other):
    return self.to_dict() == other.to_dict()
```

## The Current Solution

Check the equivalence of the dictionaries of the embeds instead of the overall objects:
```py
self.to_dict() == other.to_dict()
```

## Summary

TL;DR aren't necessarily equivalent even if their dictionaries are identical

**Comment by Rapptz** [comment]
I'm not sure there's a way to make this equality work without it being overly expensive so I'm against it purely on those grounds
